Userland/Shell/Builtin.cpp

```diff
@@ -149,6 +149,20 @@ ErrorOr<int> Shell::builtin_where(Main::Arguments arguments)
     return at_least_one_succeded ? 0 : 1;
 }
 
+ErrorOr<int> Shell::builtin_reset(Main::Arguments)
+{
+    destruct();
+    if (m_default_constructed)
+        default_construct();
+    else
+        non_default_construct(m_is_interactive);
```

Comment:
...then you can rewrite this as
```c++
    auto editor = m_editor;
    auto posix_mode = m_posix_mode;
    auto interactive = m_is_interactive;
    this->~Shell();
    new (this) Shell(editor, posix_mode);
```

It's safe to ignore the default-constructed state as you can't run anything with it.